c/parallel/src/unique_by_key.cu

```diff
@@ -0,0 +1,514 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cub/detail/choose_offset.cuh>
+#include <cub/detail/launcher/cuda_driver.cuh>
+#include <cub/device/device_select.cuh>
+
+#include <format>
+
+#include "kernels/iterators.h"
+#include "kernels/operators.h"
+#include "util/context.h"
+#include "util/indirect_arg.h"
+#include "util/scan_tile_state.h"
+#include "util/types.h"
+#include <cccl/c/unique_by_key.h>
+#include <nvrtc/command_list.h>
+#include <nvrtc/ltoir_list_appender.h>
+
+struct op_wrapper;
+struct device_unique_by_key_policy;
+using OffsetT = int64_t;
+static_assert(std::is_same_v<cub::detail::choose_signed_offset_t<OffsetT>, OffsetT>, "OffsetT must be int64");
```

Comment:
critical: both merge sort and unique by key use unsigned offset types:

```c++
  using OffsetT = detail::choose_offset_t<NumItemsT>;
```

Historicaly, when an algorithm is tested for unsigned offsets, it frequently relies on unsigned integer overflow logic to handle large problem sizes. Before we merge this PR, let's harmonize the offset type selection logic with the C++ part.

Reply:
I fixed this here and in merge sort